src/server/editorServices.ts

```diff
@@ -4056,6 +4059,106 @@ namespace ts.server {
             return false;
         }
 
+        /*@internal*/
+        requestEnablePlugin(project: Project, pluginConfigEntry: PluginImport, searchPaths: string[], pluginConfigOverrides: Map<any> | undefined) {

```

Comment:
Can you move this back to project, seems like its being called from project and most of the work is redirected to project?

Reply:
The back and forth keeps most of the logic that was originally in `Project` still in `Project`, but allows the `ProjectService` to coordinate async plugin load across all projects, as well as manage the lifetime of `pendingPluginEnablements` below.